app/models/category_list.rb

```diff
@@ -106,7 +106,10 @@ def relevant_topics_query
           )
     end
 
-    @all_topics = TopicQuery.remove_muted_tags(@all_topics, @guardian.user).includes(:last_poster)
+    inclusions = [:last_poster]
+    preload = DiscoursePluginRegistry.category_list_topics_preloader_associations
+    inclusions.concat(preload) if preload.present?
+    @all_topics = TopicQuery.remove_muted_tags(@all_topics, @guardian.user).includes(inclusions)
```

Comment:
```suggestion
    @all_topics = TopicQuery.remove_muted_tags(@all_topics, @guardian.user).includes(:last_poster)

    if (preload = DiscoursePluginRegistry.category_list_topics_preloader_associations).present?
      @all_topics.includes(preload)
    end

    @all_topics
```

Abit of code golf here but this reduces the need for two array allocation while still maintaining readability.